src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -1,47 +1,153 @@
 package core.basesyntax;
 
 import java.util.List;
+import java.util.Objects;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private int size;
+    private Node<T> head;
+    private Node<T> tail;
+
+    private class Node<T> {
+        private T element;
+        private Node<T> next;
+        private Node<T> prev;
+
+        public Node(Node<T> prev, T element, Node<T> next) {
+            this.element = element;
+            this.prev = prev;
+            this.next = next;
+        }
+    }
+
     @Override
     public void add(T value) {
+        Node<T> newNode = new Node<>(null, value, null);
+        if (head == null) {
+            head = tail = newNode;
+        } else {
+            tail.next = newNode;
+            newNode.prev = tail;
+            tail = newNode;
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        Objects.checkIndex(index, size + 1);
+        Node<T> newNode = new Node<>(null, value, null);
+        if (head == null) {
+            head = tail = newNode;
+        } else if (index == 0) {
+            newNode.next = head;
+            head.prev = newNode;
+            head = newNode;
+        } else if (index == size) {
+            tail.next = newNode;
+            newNode.prev = tail;
+            tail = newNode;
+        } else {
+            Node<T> currentNode = getElementByIndex(index - 1);
+            newNode.next = currentNode.next;
+            currentNode.next.prev = newNode;
+            currentNode.next = newNode;
+            newNode.prev = currentNode;
+        }
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (int i = 0; i < list.size(); i++) {
+            tail.next = new Node<>(tail, list.get(i), null);
+            tail = tail.next;
+            size++;
+        }
```

Comment:
use the method `add`
```suggestion
        for (int i = 0; i < list.size(); i++) {
            add(list.get(i));
        }
```